### ISSUE TITLE
Installation instructions don't seem to work for Mac OS with python 3.11, torch versions are not available from pytorch

### ISSUE THREAD

**Issue by frogeyedpeas** [Wangt-CN/DisCo#53]
Hey the versions of torch and torchvision used by this project are no longer valid see here:

```
sidharthghoshal@Sidharths-MBP Disco % pip install --user torch==1.12.1+cu113 torchvision==0.13.1+cu113 -f https://download.pytorch.org/whl/torch_stable.html

DEPRECATION: Loading egg at /usr/local/lib/python3.11/site-packages/python_qsoptex-0.5-py3.11-macosx-11-x86_64.egg is deprecated. pip 23.3 will enforce this behaviour change. A possible replacement is to use pip for package installation..
DEPRECATION: Loading egg at /usr/local/lib/python3.11/site-packages/six-1.16.0-py3.11.egg is deprecated. pip 23.3 will enforce this behaviour change. A possible replacement is to use pip for package installation..
Looking in links: https://download.pytorch.org/whl/torch_stable.html
ERROR: Could not find a version that satisfies the requirement torch==1.12.1+cu113 (from versions: 2.0.0, 2.0.1, 2.1.0)
ERROR: No matching distribution found for torch==1.12.1+cu113
```

I'm going to try installing torch version 2.0 and similarly for torchvision and see if the project doesn't break. If it does break then it is likely some new installation instructions need to be found. 

Also: Possibly related: https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/5485

**Comment by frogeyedpeas** [comment]
This might be specific to macos

**Comment by Wangt-CN** [comment]
Hi @frogeyedpeas , thanks for your reporting. The original code runs under python3.8 and pytorch 1.12.1. I think the reason may be that such pytorch version cannot be used on MacOs. Btw, did you succeed to run the code under pytorch 2.0?

**Comment by snimavat** [comment]
I am unable to install torch 2.4.1 on macpro with python 3.11

```
pip index versions torch
WARNING: pip index is currently an experimental command. It may be removed/changed in a future release without prior warning.
torch (2.2.2)
Available versions: 2.2.2, 2.2.1, 2.2.0, 2.1.2, 2.1.1, 2.1.0, 2.0.1, 2.0.0

```
